Yosemite/YosemiteTests/Stores/ProductStoreTests.swift

```diff
@@ -340,6 +340,29 @@ class ProductStoreTests: XCTestCase {
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
 
+    /// Verifies that `ProductAction.retrieveProduct` returns the expected `Product` of external product type.
+    ///
+    func testRetrieveSingleExternalProductReturnsExpectedFields() {
+        // Arrange
+        let expectation = self.expectation(description: "Retrieve single external product")
+        let remote = MockProductsRemote()
+        let expectedProduct = MockProduct().product(siteID: sampleSiteID, productID: sampleProductID, buttonText: "Deal", externalURL: "https://goodeals.com")
+        remote.whenLoadingProduct(siteID: sampleSiteID, productID: sampleProductID, thenReturn: .success(expectedProduct))
+        let productStore = ProductStore(dispatcher: dispatcher, storageManager: storageManager, network: network, remote: remote)
+
+        // Action
+        let action = ProductAction.retrieveProduct(siteID: sampleSiteID, productID: sampleProductID) { (product, error) in
+            // Assert
+            XCTAssertNil(error)
+            XCTAssertEqual(product, expectedProduct)
+
+            expectation.fulfill()
```

Comment:
Also here, I suggest the use of `waitForExpectation:`

Reply:
yup! updated in https://github.com/woocommerce/woocommerce-ios/pull/2392/commits/d78a26ed188f0d47b5c37d13f48e7c6278521b61